File: c/src/Structural/Bridge/Soda.c
0

//// Original copy of this content taken from http://www.fluffycat.com/Java-Design-Patterns/ in 2010
//// Original Author: Larry Truett
//// Privacy Policy at http://www.fluffycat.com/Privacy-Policy/
//Java Design Patterns Bridge
//Bridge Overview
//An abstraction and implementation are in different class hierarchies.
//Soda.java - the Abstract Base Class

public abstract class Soda {  
   SodaImp sodaImp; 
   
   public void setSodaImp() {
       this.sodaImp = SodaImpSingleton.getTheSodaImp();
   }
   public SodaImp getSodaImp() {
       return this.sodaImp;
   }
   
   public abstract void pourSoda();
}

File: javascript/src/Structural/Bridge/Soda.js
(function() {
  define(['Structural/Bridge/SodaImpSingleton'], function(SodaImpSingleton) {
    var Soda;
    Soda = (function() {
      Soda.prototype.sodaImp = null;

      function Soda() {
        throw "Can't instantiate abstract class, Soda";
      }

      Soda.prototype.setSodaImp = function() {
        return this.sodaImp = SodaImpSingleton.getTheSodaImp();
      };

      Soda.prototype.getSodaImp = function() {
        return this.sodaImp;
      };

      Soda.prototype.pourSoda = function() {
        throw "Unimplemented method Soda.pourSoda";
      };

      return Soda;

    })();
    return Soda;
  });

}).call(this);
